sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1875,14 +1895,22 @@ def process(self, unused_element, signal):
     temp_location = pcoll.pipeline.options.view_as(
         GoogleCloudOptions).temp_location
     gcs_location = self._get_destination_uri(temp_location)
+    job_name = pcoll.pipeline.options.view_as(GoogleCloudOptions).job_name
 
+    try:
+      step_name = self.label
```

Comment:
I looked at the file, and it seems like self.label not referenced anywhere else in the class, and I don't see a call to super ctor which would set it.

Is this using a werid pattern, where something external is taking the object and attaching parameters to it?

Is it possible to fix up that style, so there is a clear method or something we can see where it gets added extenrally?

Or to t lease define a default, empty value with a comment saying how it gets set

Reply:
This is set by the Python SDK as it expands transforms. It's definitely not orthodox to rely on it, and a little unusual.